primitives/primproc/bppsendthread.h

```diff
@@ -124,7 +141,10 @@ class BPPSendThread
 
     /* secondary queue size restriction based on byte size */
     volatile uint64_t currentByteSize;
-    uint64_t maxByteSize;
+    static uint64_t maxByteSize;
```

Comment:
Sorry but static member looks a bit hack-ish here.
AFAIK the only place where PP creates BPPSendThread is BPPV ctor(primproc/primitiveserver.cpp) and it is possible to ship the value inside PrimitiveServer which ptr is available in the BPPV ctor.

Reply:
Yes indeed we could do that. My intent was to not have to deal with (not spend cpu cycles on) that. It's not horribly expensive, but I am trying to shave where I can.
The drawback of my approach is that a restart is definitely required to notice a change to the xml setting.

Reply:
I don't suffer from a prejudice against static members:)

Reply:
I do b/c it represents a state variable that lives outside the scope of the code that uses it. Nobody guarantees anything about such states. It is hard to maintain, hard to follow, hard to debug.

Reply:
So the options are to create a unscoped global, or to spend cpu to unnecessarily re-initialize in the constructor or to call joblist::ResourceManager::instance()->getMaxBPPSendQueue() each time we need the value.

Reply:
I find it easier to maintain and follow static than global variables. At least as a static, it's usage is clearly (without an accesor) for use by class members.

Reply:
I could add const, if it helps.
If I just remove it from the class declaration, it still works as a global. The reason I put it in the class is for documentation (to make it clear it's use is for and by the class.
Not sure why you find this confusing.

Reply:
I don't see any reason against moving config access from getMaxBPPSendQueue() code into RM ctor, so that RM::getMaxBPPSendQueue() just reads the value from RM singleton. Finally you can reference RM::getMaxBPPSendQueue() in the BPPSendThread ctor code in bppsendthread.cpp. What do you think about this approach?

Reply:
Changed to your recommendation, though I still feel (even with the change to ResourceManager) that a static makes more sense. But for the sake of moving forward, I put it in the sendthread ctor.